contracts/extensions/ReputationBootstrapper.sol

```diff
@@ -137,11 +145,27 @@ contract ReputationBootstrapper is ColonyExtensionMeta {
     }
   }
 
+  /// @notice Commit the secret, beginning the security delay window
+  /// @param _committedSecret A sha256 hash of (userAddress, secret)
+  function commitSecret(bytes32 _committedSecret) public locked {
+    committedSecrets[_committedSecret] = block.timestamp;
```

Comment:
Unfortunately an attacker can call `commitSecret` with the committed (good) secret after the user has broadcast the `claimGrant` transaction and prevent them from claiming indefinitely, allowing the attacker to claim at their leisure. This mapping needs to be based on `msgSender()` to prevent that, I think? I guess to future-proof ourselves in case a user needs to go through this process multiple times, it can be based on `keccak(msgSender(), committedSecret` to allow them to do so in parallel.

Reply:
Gotcha. I had initially thinking about including `msgSender` but didn't as the hash includes `msgSender`, but I see your point about the griefing. In what case do we need to future-proof, assuming each secret can be claimed only once?